pkg/upgraders/healthcheckstep.go

```diff
@@ -23,80 +20,29 @@ func (c *clusterUpgrader) PreUpgradeHealthCheck(ctx context.Context, logger logr
 		return true, nil
 	}
 
-	ok, err := performClusterHealthCheck(c.metrics, c.cvClient, c.config, logger)
+	ok, err := CriticalAlerts(c.metrics, c.config, c.upgradeConfig, logger)
 	if err != nil || !ok {
-		c.metrics.UpdateMetricClusterCheckFailed(c.upgradeConfig.Name)
 		return false, err
 	}
 
-	c.metrics.UpdateMetricClusterCheckSucceeded(c.upgradeConfig.Name)
-	return true, nil
-}
-
-// PostUpgradeHealthCheck performs cluster healthy check
-func (c *clusterUpgrader) PostUpgradeHealthCheck(ctx context.Context, logger logr.Logger) (bool, error) {
-	ok, err := performClusterHealthCheck(c.metrics, c.cvClient, c.config, logger)
+	ok, err = ClusterOperators(c.metrics, c.cvClient, c.upgradeConfig, logger)
```

Comment:
No problem with the code. But on the logic point of view, do we need to check all the healthy condition?
In this implementation, if there is any critical alert, code will return then there is no check on the cluster operators.

Reply:
Thanks Tony! Yeah, nice point. For now, the checks are blocking in nature and we are maintaining that. So if there's any health check failing then upgrade will not proceed. However, we will classify these failures better and categorize into blocking/non-blocking alerts/issues and notify accordingly. For now, the health check continues to remain similar as in https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/upgraders/healthcheckstep.go#L51-L101 in this refactoring PR. In future PR we will gather these failures together and notify once. Let me know if this answers your query.

Reply:
I understand that any block alerts/issues will make the upgrade not be processed. My question is there is a chance that both blocking alert and blocking degraded cluster operator exist. In this case, the alert check will fail then code returns, the degraded operators will not be checked. Then only alert failure has been notified. Customer may not notice there is also blocking degraded cluster operator.
Let me know if I misunderstand something.